c++/src/kj/compat/tls.c++

```diff
@@ -445,8 +459,58 @@ public:
   }
 
 private:
+  Promise<void> doOneAccept() {
+    // Accept one connection and queue up the next accept on our TaskSet
+
+    auto loopAgainOnSuccess = [this](AuthenticatedStream&& stream) {
+      auto promise = kj::evalNow([&] {
+        // Do the SSL acceptance procedure.
+        return tls.wrapServer(kj::mv(stream));
+      });
+
+      auto pushStream = [this](AuthenticatedStream&& stream) -> Promise<void> {
+        // This is only attached to the success path, thus the error handler will catch if our
+        // promise fails.
+        queue.push(kj::mv(stream));
+        return kj::READY_NOW;
+      };
+      tasks.add(promise.then(kj::mv(pushStream)));
+
+      // Queue up the next accept loop immediately without waiting for SSL_accept()/wrapServer().
+      tasks.add(doOneAccept());
+    };
+
+    auto loopAgainOnFailure = [this](Exception&& e) {
+      KJ_LOG(ERROR, "TlsConnectionReceiver could not accept()", e);
```

Comment:
I'd argue that if someone is actually listening using `onFatalError()`, we shouldn't log, since it should be up to the caller to decide if and how they want to log the error. But see my comments on whether `onFatalError()` is the right interface in the first place...